### PR TITLE
Dashboard to have button to shrink and expand source classes

### LINKED ISSUE #437: Zip source classes in dashboard
Shrink/expand control via JavaScript (without a framework).

image

Make close/expand when the item is more than 5.

Design

A button shows "close" when it's shown.

The button shows "expand" button when the content is hidden.

If the number of item is more than or equal to 5, it's hidden at first.

The button

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #437. (As part of #395)

image

### CODE REVIEW

File: dashboard/src/main/resources/js/dashboard.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+function toggleSourceClassListVisibility(button) {
+  const classList = button.parentElement.nextElementSibling;
```
Comment (suztomo) [inline dashboard/src/main/resources/js/dashboard.js]: Google JavaScript style uses `const`
https://google.github.io/styleguide/jsguide.html#features-use-const-and-let

File: dashboard/src/main/resources/templates/macros.ftl [hunk 2]
```diff
@@ -16,9 +16,17 @@
       ${pluralize(sourceClassCount, "source class", "source classes")}.
     </p>
     <#list causeToSourceClasses.keySet() as errorCause >
-      <p class="jar-linkage-report-cause">${errorCause?html}, referenced from</p>
```
Comment (suztomo) [inline dashboard/src/main/resources/templates/macros.ftl]: This part will need to merge the other PR (#431 )

File: dashboard/src/main/resources/templates/component.ftl [hunk 3]
```diff
@@ -3,6 +3,7 @@
   <head>
     <title>Google Cloud Platform Dependency Analysis Report for ${groupId}:${artifactId}:${version}</title>
     <link rel="stylesheet" type="text/css" href="dashboard.css" />
+    <script src="dashboard.js"></script>
```
Comment (suztomo) [inline dashboard/src/main/resources/templates/component.ftl]: Google html style does not specify type attribute https://google.github.io/styleguide/htmlcssguide.html#type_Attributes

File: dashboard/src/main/resources/js/dashboard.js (hunk 1 above)
Comment (elharo) [reply dashboard/src/main/resources/js/dashboard.js]: something like these for the symbols (not + and -) 

image

Try Unicode character U+25B6 for a closed zippy icon and U+25BC for an open zippy icon.

**Comment by suztomo** [comment]
@elharo Updated the icon. PTAL.

File: dashboard/src/main/resources/templates/component.ftl [hunk 4]
```diff
@@ -1,8 +1,11 @@
+<!DOCTYPE html>
```
Comment (suztomo) [inline dashboard/src/main/resources/templates/component.ftl]: HTML 5. https://google.github.io/styleguide/htmlcssguide.html#Document_Type

File: dashboard/src/main/resources/templates/component.ftl [hunk 5]
```diff
@@ -1,8 +1,11 @@
...
 <html lang="en-US">
   <#include "macros.ftl">
   <head>
+    <meta charset="utf-8" />
```
Comment (suztomo) [inline dashboard/src/main/resources/templates/component.ftl]: Without encoding specification, the multi-byte character "▶" was not shown correctly.

Although Google HTML style says [do not  close void elements](https://google.github.io/styleguide/htmlcssguide.html#Document_Type), meta tags need to close in order to be parsed by xom document parser for unit tests.

**Pull request merged**